devs/docs/release.rst

```diff
@@ -0,0 +1,88 @@
+======================
+Git & Release workflow
+======================
+
+- Development happens in feature branches. Everything that goes into master
+  must be reviewed.
+
+- PRs should be small but self-contained. "Dead code" (Features that aren't
+  integrated) cannot go into master.
+
+- Commits must be self-contained. Each commit must compile and tests should
+  pass. (Flaky tests happen.)
```

Comment:
"Flaky tests happen" sounds like they are acceptable. How about "Beware of Flaky tests"?